struct-patch-derive/src/lib.rs

```diff
@@ -100,6 +100,27 @@ impl Patch {
         #[cfg(not(feature = "status"))]
         let patch_status_impl = quote!();
 
+        let add_impl = quote! {
+            impl core::ops::Add<#name #generics> for #struct_name #generics #where_clause {
+                type Output = #struct_name;
+
+                fn add(mut self, rhs: #name #generics) -> Self {
+                    self.apply(rhs);
+                    self
+                }
+            }
+
+            impl core::ops::Add<#struct_name #generics> for #name #generics #where_clause {
+                type Output = #struct_name;
+
+                fn add(mut self, rhs: #struct_name #generics) -> #struct_name  #where_clause {
+                    let mut rhs = rhs;
+                    rhs.apply(self);
+                    rhs
+                }
+            }
+        };
+
```

Comment:
I think that this should be in a flag since the lib is `no_std` compatible, it would be better to do not need the `core` lib if not needed:

```suggestion
        #[cfg(feature = "add")]
        let add_impl = quote! {
            impl core::ops::Add<#name #generics> for #struct_name #generics #where_clause {
                type Output = #struct_name;

                fn add(mut self, rhs: #name #generics) -> Self {
                    self.apply(rhs);
                    self
                }
            }

            impl core::ops::Add<#struct_name #generics> for #name #generics #where_clause {
                type Output = #struct_name;

                fn add(mut self, rhs: #struct_name #generics) -> #struct_name  #where_clause {
                    let mut rhs = rhs;
                    rhs.apply(self);
                    rhs
                }
            }
        };
        #[cfg(not(feature = "add"))]
        let add_impl = quote!();

```

Reply:
I think we can do `no_std`, but we need to test before we say it.